.build/config/cassandra_ci.yaml

```diff
@@ -0,0 +1,342 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Contains definitions of all pipelines and jobs (test suites) in Apache Cassandra's CI.
+#
+
+# We have 2 concepts in CI:
+#   1. Pipelines: a collection of test suites to be run
+#   2. Jobs: a set of tests to be run with a corresponding command that executes said testing
+#
+# We also have a variety of parameterizable settings in tests that need to be exercised to confirm different configurations
+# both work in isolation and work when combined with other params.
+
+
+#-----------------------------------------------------------------------------
```

Comment:
@dcapwell - could I get you to take another look at the new .yaml structure here?

Changes:
- pipelines
- jobs
- runtime_env (i.e. base params and env for collection of jobs)
- parameters (for matrixing on post-commit)
- added timeouts and multiplex values here